runtime/onert/backend/train/MemoryPlanner.test.cc

```diff
@@ -25,9 +27,74 @@ using onert::ir::OperandIndex;
 using onert::ir::OperationIndex;
 
 // TODO: Add test testcase for {Bump, FirstFit, WIC}Planner<LayerScopeTensor>
+namespace
+{
+
+template <typename T> T to_index(uint32_t first, uint32_t second) = delete;
+
+template <> DisposableTensorIndex to_index<DisposableTensorIndex>(uint32_t first, uint32_t second)
+{
+  return DisposableTensorIndex{OperationIndex{first}, OperandIndex{second}};
+}
 
-TEST(BumpPlanner, claim_test)
+template <> LayerScopeTensorIndex to_index<LayerScopeTensorIndex>(uint32_t first, uint32_t second)
 {
+  return LayerScopeTensorIndex{OperationIndex{first}, second};
+}
+
+template <template <typename> class Planner, typename Index> class PlannerVerifier final
+{
+public:
+  void claim(uint32_t first_idx, uint32_t second_idx, size_t size)
+  {
+    auto index = to_index<Index>(first_idx, second_idx);
+    _planner.claim(index, size);
+  }
+
+  // Claim plan and verify newly added plan, call ASSERT_* on failure
+  void claim(uint32_t first_idx, uint32_t second_idx, size_t size, uint32_t expected_offset)
+  {
+    auto index = to_index<Index>(first_idx, second_idx);
+    _planner.claim(index, size);
+
+    auto mem_blk = _planner.memory_plans()[index];
+    ASSERT_EQ(mem_blk.offset, expected_offset);
+    ASSERT_EQ(mem_blk.size, size);
+  }
+
+  void release(uint32_t first_idx, uint32_t second_idx)
+  {
+    _planner.release(to_index<Index>(first_idx, second_idx));
+  }
+
+  // Verify capacity, call ASSERT_* on failure
+  void capacity(uint32_t expected_capacity)
+  {
+    auto actual_capacity = _planner.capacity();
+    ASSERT_EQ(actual_capacity, expected_capacity);
+  }
+
+  // Verify memory_plans's size and offset, calls ASSERT_* on failure
+  void verify(uint32_t first_idx, uint32_t second_idx, size_t expected_size,
+              uint32_t expected_offset)
+  {
+    auto index = to_index<Index>(first_idx, second_idx);
+    auto mem_blk = _planner.memory_plans()[index];
+    ASSERT_EQ(mem_blk.offset, expected_offset);
+    ASSERT_EQ(mem_blk.size, expected_size);
+  }
+
+private:
+  Planner<Index> _planner;
+};
+
+} // namespace
+
+TEST(BumpPlanner, disposable_claim_test)
+{
+  PlannerVerifier<BumpPlanner, DisposableTensorIndex> p;
+  /*
+  TODO - remove commented code
```

Comment:
(Optional)
Why didn't you remove the code?

Reply:
If I remove it, the diff page(https://github.com/Samsung/ONE/pull/13974/files) is messed up. 
I was afraid that checking - the test values have not changed at all - becomes difficult.